src/main/java/org/gridsuite/filter/server/AbstractFilterRepositoryProxy.java

```diff
@@ -43,12 +43,35 @@ static SortedSet<String> setToSorterSet(Set<String> set) {
         return CollectionUtils.isEmpty(set) ? null : new TreeSet<>(set);
     }
 
+    static Map<String, Set<String>> convert(FreePropertiesFilterEntity entity) {
+        if (entity == null) {
+            return null;
+        }
+
+        Map<String, Set<String>> ret = entity.getFreePropertyFilterEntities().stream()
+            .collect(Collectors.toMap(FreePropertyFilterEntity::getPropName, FreePropertyFilterEntity::getPropValues));
+
+        return ret;
+    }
+
     static NumericFilterEntity convert(NumericalFilter numericalFilter) {
         return numericalFilter != null ?
                 new NumericFilterEntity(null, numericalFilter.getType(), numericalFilter.getValue1(), numericalFilter.getValue2())
                 : null;
     }
 
+    static FreePropertiesFilterEntity convert(Map<String, Set<String>> dto) {
+        if (dto == null)  {
+            return null;
+        }
+
+        Set<FreePropertyFilterEntity> innerEntities = dto.entrySet().stream()
+            .map(p -> FreePropertyFilterEntity.builder()
+                .propName(p.getKey()).propValues(p.getValue()).build()).collect(Collectors.toSet());
+        FreePropertiesFilterEntity mapEntities = FreePropertiesFilterEntity.builder().freePropertyFilterEntities(innerEntities).build();
+        return mapEntities;
```

Comment:
you can do here also
return FreePropertiesFilterEntity.builder().freePropertyFilterEntities(innerEntities).build();

Reply:
done too